src/pipeline.py

```diff
@@ -0,0 +1,78 @@
+import multiprocessing as mp
+import os
+import shutil
+
+from common import get_files, get_log
+from extract import extract
+from ingest import ingest
+from parse import parse
+from standardize import standardize
+
+log = get_log(__file__)
+
+
+def clean_dirs(list_of_dirs):
+    """
+    Deletes and recreates empty directories
+    """
+    for dir_ in list_of_dirs:
+        try:
+            shutil.rmtree(dir_)
+        except FileNotFoundError:
+            pass
+        os.makedirs(dir_)
+
+
+def get_filename_without_extension(filename):
+    return os.path.basename(filename).split(".")[0]
+
+
+def get_pipeline_files(raw_dir, extracted_dir, parsed_dir, standardized_dir):
+    """
+    yields file names corresponding to the raw, extracted, parsed, standardized
+    intermediate steps pf the pipeline
+    """
+    suffix = ".json"
+    for raw_file in get_files(raw_dir):
+        filestem = get_filename_without_extension(raw_file)
+        extracted_file = os.path.join(extracted_dir, filestem + suffix)
+        parsed_file = os.path.join(parsed_dir, filestem + suffix)
+        standardized_file = os.path.join(standardized_dir, filestem + suffix)
+        yield raw_file, extracted_file, parsed_file, standardized_file
+
+
+def _etl(raw, extracted, parsed, standardized):
+    extract(raw, extracted)
+    parse(extracted, parsed)
+    standardize(parsed, standardized)
+
+
+def run(data_dir):
+    cores = mp.cpu_count()
+    pool = mp.Pool(cores)
+    jobs = []
+
+    raw_dir = os.path.join(data_dir, "raw")
+    extracted_dir = os.path.join(data_dir, "extracted")
+    parsed_dir = os.path.join(data_dir, "parsed")
+    standardized_dir = os.path.join(data_dir, "standardized")
+
+    clean_dirs([extracted_dir, parsed_dir, standardized_dir])
```

Comment:
Do you think it might be useful to have this as a toggleable flag? I'm a bit terrified of yolo deleting directories without prompts (even though I realize this is mostly internal). Perfectly fine if you think that this is never meant to be seen by the end user and is okay to delete / clean.

Reply:
That's a good point. Maybe a safer pattern we could do is use a `tempdir` for the pipeline steps and do a 1-way copy into the internal dirs of this app (for inspection) and not yolo delete directories 😄

Reply:
Changed this now to `make_dirs` , no more rmtree. Files within the extracted, parsed, and standardized directories will still get overridden if they exist there with the same name, but nothing else will get touched.